src/datadog/config_manager.cpp

```diff
@@ -1,80 +1,164 @@
 #include "config_manager.h"
 
+#include <sstream>
+
+#include "parse_util.h"
 #include "trace_sampler.h"
 
 namespace datadog {
 namespace tracing {
+namespace {
+
+std::string join(const std::vector<StringView>& values,
+                 const char* const separator) {
+  if (values.empty()) return {};
+  auto it = values.cbegin();
+
+  std::string res{*it};
+  for (++it; it != values.cend(); ++it) {
+    res += separator;
+    append(res, *it);
+  }
+
+  return res;
+}
+
+// TODO: use `to_chars`
+std::string to_string(double d) {
+  std::stringstream stream;
+  stream << std::fixed << std::setprecision(1) << d;
+  return stream.str();
+}
```

Comment:
This code is similar to that in `tracer_config.cpp`. Perhaps all of these should be in a dedicated component. One advantage would be knowing there's one place to look for `to_string` and `join` overloads, as opposed to wondering "is it a local utility, or declared in some header I included?"

Reply:
addressed in e2bb8f465b2245055cb150d6d5bc8bcf69211c65